### PR TITLE
chore: add more tests for empty and non-present  

### PR DESCRIPTION
**Pull request by tobiasdiez**
Add a few more test cases around empty and non-present bodies for https://github.com/unjs/h3/issues/197.

Some of the tests are currently failing as readBody/readBodyRaw for a request with `Content-Type: 'text/plain'` and `body: ''`  return undefined now. Is this the expected behavior? #197 was about distinguishing empty vs non-present bodies, but both return undefined now.

### CODE REVIEW

File: test/body.test.ts [hunk 1]
```diff
@@ -48,14 +84,34 @@ describe('', () => {
...
       expect(result.text).toBe('200')
     })
 
+    it('handles empty body', async () => {
+      let _body = 'initial'
+      app.use('/', eventHandler(async (request) => {
+        _body = await readBody(request); return '200'
+      }))
+      const result = await request.post('/api/test').set('Content-Type', 'text/plain').send('')
+      expect(_body).toStrictEqual({})
```
Comment (pi0) [inline test/body.test.ts:101]: Now i see what you mean 👍🏼  This seems a wrong behavior from`h3.readBody` that empty string results in empty object.

File: test/body.test.ts [hunk 2]
```diff
@@ -26,6 +26,42 @@ describe('', () => {
...
+      expect(result.text).toBe('200')
+    })
+
+    it('returns an empty string if body is empty', async () => {
+      let body = 'initial'
+      app.use('/', eventHandler(async (request) => {
+        body = await readRawBody(request) as string
+        return '200'
+      }))
+      const result = await request.post('/api/test').send('')
```
[the remark this answers is not in the document]
Comment (tobiasdiez) [reply test/body.test.ts]: Okay, this seem to work indeed.

**Comment by pi0** [comment]
Thanks <3
